llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -2569,6 +2569,17 @@ MVT AArch64TargetLowering::getScalarShiftAmountTy(const DataLayout &DL,
 bool AArch64TargetLowering::allowsMisalignedMemoryAccesses(
     EVT VT, unsigned AddrSpace, Align Alignment, MachineMemOperand::Flags Flags,
     unsigned *Fast) const {
+
+  // Allow SVE loads/stores where the alignment >= the size of the element type,
+  // even with +strict-align. The SVE loads/stores do not require memory to be
+  // aligned more than the element type even without unaligned accesses.
+  // Without this, already aligned loads and stores are forced to have 16-byte
+  // alignment, which is unnecessary and fails to build as
+  // TLI.expandUnalignedLoad() and TLI.expandUnalignedStore() don't yet support
+  // scalable vectors.
```

Comment:
Rewritten the comment. Is it safe to assume the `str z*` is only used for spills/fills (and possibly via an intrinsic, which I don't think uses these alignment checks)?

Reply:
For scalable data-vectors ldr/str is only ever used for spills and fills. For predicate regs, there is only str/ldr and so their addresses must be at least 2-byte aligned.

Reply:
We should avoid do anything to prevent an expanded use of ldr/str though.  I know I've toyed with it in the past as a way to remove predicate data dependencies.

Reply:
I don't think this prevents an expanded use of ldr/str for operations that are known to be 16-byte aligned. For operations with < 16-byte alignment any expansion in expandUnalignedLoad/Store would negate any benefit of using ldr/str over predicated loads/stores over anyway. 

I don't think inline asm is relevant here, I think these hooks are mainly used things like `StoreSDNode`/`LoadSDNode`.